src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5595,6 +5671,41 @@ void SessionImpl::handleTorrentDeleteFailedAlert(const lt::torrent_delete_failed
     m_removingTorrents.erase(removingTorrentDataIter);
 }
 
+void SessionImpl::handleTorrentNeedCertAlert(const lt::torrent_need_cert_alert *p)
+{
+    const TorrentID torrentID {p->handle.info_hash()};
+
+    TorrentImpl *const torrent = m_torrents.value(torrentID);
+    if (!torrent) [[unlikely]]
+        return;
+
+    const Path baseDir = sslCertificatesDirectory();
+    if (baseDir.exists())
+    {
+        auto [certPath, keyPath, dhPath] = sslCertificatesPathsForTorrent(torrentID);
+
+        if (!certPath.exists() || !keyPath.exists())
+        {
+            certPath = baseDir / Path(u"default.crt.pem"_s);
+            keyPath = baseDir / Path(u"default.key.pem"_s);
+        }
+        if (!dhPath.exists())
+        {
+            dhPath = baseDir / Path(u"default.dh.pem"_s);
+        }
+
+        if (certPath.exists() && keyPath.exists() && dhPath.exists())
+        {
+            torrent->nativeHandle().set_ssl_certificate(certPath.toString().toStdString()
+                    , keyPath.toString().toStdString()
+                    , dhPath.toString().toStdString());
+            return;
+        }
+    }
+
+    LogMsg(tr("Unable to set certificates for SSL torrent \"%1\" ").arg(torrentID.toString()));
```

Comment:
```suggestion
    LogMsg(tr("Unable to set SSL certificates for torrent. Torrent: \"%1\". Reason: \"%2\"").arg(torrent->name(), QString::fromStdString(p->message())), Log::WARNING);
```

Reply:
I believe `p->message()` is irrelevant here.

Reply:
>I believe p->message() is irrelevant here.

Why would it be irrelevant?

Reply:
> Why would it be irrelevant?

Because it is not the reason of "Unable to set SSL certificates for torrent". It is message from "torrent_need_cert_alert" so it should be about the fact that torrent needs SSL sertificate.

Reply:
Ok, ignore the `p->message()` part.